lib/elixir/lib/string.ex

```diff
@@ -1894,15 +1939,13 @@ defmodule String do
 
   """
   @spec last(t) :: grapheme | nil
-  def last(string) when is_binary(string) do
-    do_last(next_grapheme(string), nil)
-  end
+  def last(""), do: nil
+  def last(string) when is_binary(string), do: do_last(:unicode_util.gc(string), nil)
 
-  defp do_last({char, rest}, _) do
-    do_last(next_grapheme(rest), char)
-  end
-
-  defp do_last(nil, last_char), do: last_char
+  defp do_last([gc | rest], _), do: do_last(:unicode_util.gc(rest), gc)
+  defp do_last([], acc) when is_binary(acc), do: acc
+  defp do_last([], acc), do: :unicode.characters_to_binary([acc])
+  defp do_last({:error, <<byte, rest::bitstring>>}, _), do: do_last(rest, <<byte>>)
```

Comment:
```suggestion
  defp do_last({:error, <<byte, rest::bitstring>>}, _), do: do_last(:unicode_util.gc(rest), <<byte>>)
```

Reply:
Awesome find. Fixed and regression test added.